pkg/auth/dependency/oauth/store_grant.go

```diff
@@ -0,0 +1,25 @@
+package oauth
+
+type CodeGrantStore interface {
+	GetCodeGrant(codeHash string) (*CodeGrant, error)
+	CreateCodeGrant(*CodeGrant) error
+	UpdateCodeGrant(*CodeGrant) error
+	DeleteCodeGrant(*CodeGrant) error
+}
+
+type OfflineGrantStore interface {
+	GetOfflineGrant(id string) (*OfflineGrant, error)
```

Comment:
Offline grants are very similar to sessions, that session APIs are able to manipulate them. Therefore its implementation is very much like session's, that it can be get using ID (and token hash is verified separately by consumer). The refresh token would be encoded like session, i.e. offline grant ID + grant token hash.